File: src/main/scala/org/brzy/webapp/controller/Secured.scala
/*
 * Copyright 2010 Michael Fortin <mike@brzy.org>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");  you may not use this
 * file except in compliance with the License. You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
 * language governing permissions and limitations under the License.
 */
package org.brzy.webapp.controller

import org.brzy.webapp.action.Roles

/**
 * Marks a controller as a secure controller using role based authentication.
 * 
 * @author Michael Fortin
 */
trait Secured extends Intercepted {

  /**
   * Sets the roles default allowed for the controller.
   */
  val roles = Roles("ADMIN","USER")
}

File: mods/brzy-security/src/main/java/org/brzy/security/Secured.java
package org.brzy.security;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Document Me..
 *
 * @author Michael Fortin
 * @version $Id: $
 */
@Retention(RetentionPolicy.RUNTIME)
public @interface Secured {
    String[] value();
}